### PR TITLE
[FEATURE] ID/PK `ColumnPairExpectations` and `MultiColumnMapExpectations` - Spark

### PR DESCRIPTION
**Pull request by Shinnnyshinshin**
# What is this PR? 
- Enables ID/PK for `ColumnPairExpectations` and `MultiColumnMapExpectations` by.
   1. Adding `UNEXPECTED_INDEX_LIST` and `UNEXPECTED_INDEX_QUERY` to validation dependencies
   2. Modifying metric functions to take in domain column **list** rather than just domain column. 
- Enables JSON-defined tests to run against Spark for ID/PK
- Adds tests at Checkpoint level for MultiColumn and ColumnPair Expectations

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: great_expectations/self_check/util.py [hunk 1]
```diff
@@ -3065,15 +3064,15 @@ def check_json_test_result(  # noqa: C901 - 52
                 assert result["result"]["observed_value"] in value
 
             elif key == "unexpected_index_list":
-                if isinstance(data_asset, (SqlAlchemyDataset, SparkDFDataset)):
+                if isinstance(data_asset, SqlAlchemyDataset):
                     pass
-                elif isinstance(data_asset, (SqlAlchemyBatchData, SparkDFBatchData)):
+                elif isinstance(data_asset, SqlAlchemyBatchData):
```
Comment (Shinnnyshinshin) [inline great_expectations/self_check/util.py:3069]: This enables Spark ID/PK tests for all Expectations defined as JSON files under `test_definitions`

Comment (Shinnnyshinshin) [reply great_expectations/self_check/util.py:3069]: This is also the reason why 2 tests had to be adjusted below. They were basically not being tested before (at least not at the `unexpected_index_list`-level

File: tests/test_definitions/column_map_expectations/expect_column_values_to_be_between_tz_informed.json [hunk 2]
```diff
@@ -51,7 +51,7 @@
       },
       {
         "title": "basic_negative_test_with_timestamps_tz_informed",
-        "suppress_test_for": ["trino", "bigquery", "mssql"] ,
+        "suppress_test_for": ["trino", "bigquery", "mssql", "spark"] ,
         "exact_match_out": false,
         "out": {
           "unexpected_index_list": [{"ts": "1970-01-01T12:00:01+05:30", "pk_index": 0}, {"ts": "2019-09-01T15:27:05.345678+09:00", "pk_index": 8}, {"ts": "2001-06-01T16:00:00.132678-08:00", "pk_index": 9}],
```
Comment (Shinnnyshinshin) [inline tests/test_definitions/column_map_expectations/expect_column_values_to_be_between_tz_informed.json:57]: `spark` was not being tested at the `tz-informed` level before.

File: great_expectations/expectations/metrics/map_metric_provider.py [hunk 3]
```diff
@@ -2829,26 +2829,35 @@ def _spark_map_condition_index(
     if unexpected_condition is None:
         return None
 
-    if "column" not in accessor_domain_kwargs:
```
Comment (Shinnnyshinshin) [inline great_expectations/expectations/metrics/map_metric_provider.py:2832]: We should add this back

File: tests/test_definitions/column_map_expectations/expect_column_values_to_not_be_null.json [hunk 4]
```diff
@@ -40,39 +40,36 @@
           "title": "negative_all_missing_values",
           "exact_match_out": false,
           "in": {
-            "column": "all_null",
-            "result_format": "COMPLETE"
```
Comment (Shinnnyshinshin) [inline tests/test_definitions/column_map_expectations/expect_column_values_to_not_be_null.json:44]: this `result_format` was overriding the other setting.. had to be removed.

**Review by anthonyburdi — APPROVED** [review]
:tada:

**Pull request merged**